paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -1463,68 +1369,84 @@ def wait_for_deployment(
             ),
             level="event",
         )
-        return
+        return None
 
     print(
         "Waiting for deployment of {} for '{}' to complete...".format(
             git_sha, deploy_group
         )
     )
 
-    deadline = time.time() + timeout
-    if green_light is None:
-        green_light = Event()
-    green_light.set()
-
     with progressbar.ProgressBar(maxval=total_instances) as bar:
-        while time.time() < deadline:
-            _query_clusters(clusters_data, green_light)
-            if not green_light.is_set():
-                raise KeyboardInterrupt
+        instance_done_futures = []
+        with concurrent.futures.ThreadPoolExecutor() as pool:
```

Comment:
nit: Same thing here, can `pool` be renamed as `executor` to minimize confusion?